3rdParty/LightPcapNg/LightPcapNg/src/light_pcapng.c

```diff
@@ -85,56 +85,55 @@ static struct _light_option *__parse_options(uint32_t **memory, const int32_t ma
 /// <param name="current">Block pointer with type and length filled out</param>
 /// <param name="local_data">Pointer to data which constitutes block body</param>
 /// <param name="block_start">Pointer to the start of the block data</param>
-void parse_by_block_type(struct _light_pcapng *current, const uint32_t *local_data, const uint32_t *block_start)
-{
-   switch (current->block_type)
-   {
-      case LIGHT_SECTION_HEADER_BLOCK:
-      {
-         DPRINT_HERE(LIGHT_SECTION_HEADER_BLOCK);
-         struct _light_section_header *shb = calloc(1, sizeof(struct _light_section_header));
-         struct _light_option *opt = NULL;
-         uint32_t version;
-         int32_t local_offset;
-
-         shb->byteorder_magic = *local_data++;
-         // TODO check byte order.
-         version = *local_data++;
-         shb->major_version = version & 0xFFFF;
-         shb->minor_version = (version >> 16) & 0xFFFF;
-         shb->section_length = *((uint64_t*)local_data);
-         local_data += 2;
-
-         current->block_body = (uint32_t*)shb;
-         local_offset = (size_t)local_data - (size_t)block_start;
-         opt = __parse_options((uint32_t **)&local_data, current->block_total_length - local_offset - sizeof(current->block_total_length));
-         current->options = opt;
-      }
-      break;
-
-      case LIGHT_INTERFACE_BLOCK:
-      {
-         DPRINT_HERE(LIGHT_INTERFACE_BLOCK);
-         struct _light_interface_description_block *idb = calloc(1, sizeof(struct _light_interface_description_block));
-         struct _light_option *opt = NULL;
-         uint32_t link_reserved = *local_data++;
-         int32_t local_offset;
-
-         idb->link_type = link_reserved & 0xFFFF;
-         idb->reserved = (link_reserved >> 16) & 0xFFFF;
-         idb->snapshot_length = *local_data++;
-         current->block_body = (uint32_t*)idb;
-         local_offset = (size_t)local_data - (size_t)block_start;
-         opt = __parse_options((uint32_t **)&local_data, current->block_total_length - local_offset - sizeof(current->block_total_length));
-         current->options = opt;
-      }
-      break;
+void parse_by_block_type(struct _light_pcapng *current, const uint32_t *local_data, const uint32_t *block_start) {
+    switch (current->block_type) 
+    {
+        case LIGHT_SECTION_HEADER_BLOCK: 
+		{ // PCPP patch
+            DPRINT_HERE(LIGHT_SECTION_HEADER_BLOCK);
+            struct _light_section_header *shb = calloc(1, sizeof(struct _light_section_header));
+            struct _light_option *opt = calloc(1, sizeof(struct _light_option));
```

Comment:
Doesn't that esentially leak the memory as `opt` is overwritten on L109?

Reply:
Yes you are correct for this here can be used free (opt) Memory Before Reassignment
I had created Commit below

Reply:
Why is it allocated here actually, if it will be freed in L110? I don't see any usages of the pointer in between?

Reply:
I had updated Recent commit